homework/index.js

```diff
@@ -21,27 +21,117 @@
+    createAndAppend('td', repositoryTitle, { html: 'Repository:', class: 'label' });
+    createAndAppend('a', repositoryTitle, {
+      target: '_blank',
+      href: repository.html_url,
+      html: repository.name,
+    });
+    if (repository.description !== null) {
+      const description = createAndAppend('tr', tableBody);
+      createAndAppend('td', description, {
+        html: 'Description:',
+        class: 'label',
+      });
+      createAndAppend('td', description, { html: repository.description });
+    }
+    const forks = createAndAppend('tr', tableBody);
+    createAndAppend('td', forks, { html: 'Forks:', class: 'label' });
+    createAndAppend('td', forks, { html: repository.forks });
+    const update = createAndAppend('tr', tableBody);
+    createAndAppend('td', update, { html: 'Updated:', class: 'label' });
+    createAndAppend('td', update, { html: repository.updated_at });
+  }
+
+  function showContributors(rightDiv, repository) {
+    rightDiv.innerHTML = '';
+    createAndAppend('p', rightDiv, {
+      html: 'Contributions',
+      class: 'contributor-head',
+    });
+    fetchJSON(repository.contributors_url, (err, contributors) => {
+      if (err) {
+        createAndAppend('div', rightDiv, {
+          html: err.message,
+          class: 'alert-error',
+        });
+      }
+
+      contributors.map(contributor => {
+        const contributorUL = createAndAppend('ul', rightDiv, {
+          class: 'contributor-ul',
+        });
+        const contributorLi = createAndAppend('li', contributorUL, {
+          class: 'contributor-li',
+        });
+
+        const contributorLink = createAndAppend('a', contributorLi, {
+          target: '_blank',
+          href: contributor.html_url,
+        });
+        createAndAppend('img', contributorLink, {
+          class: 'contributor-image',
+          src: contributor.avatar_url,
+        });
+        const contributorDetails = createAndAppend('div', contributorLink, {
+          class: 'contributor-details',
+        });
+
+        createAndAppend('div', contributorDetails, {
+          html: contributor.login,
+          class: 'contributor-names',
+        });
+        createAndAppend('div', contributorDetails, {
+          html: contributor.contributions,
+          class: 'contributors',
+        });
+      });
+    });
+  }
   function main(url) {
+    const root = document.getElementById('root');
+    const header = createAndAppend('header', root, { class: 'header' });
+    createAndAppend('p', header, { html: 'HYF Repositories', class: 'hyf' });
+    const select = createAndAppend('select', header, {
+      class: 'select',
+      id: 'list',
+    });
+    const container = createAndAppend('div', root, { class: 'container' });
+    const left = createAndAppend('div', container, { class: 'left_div' });
+    const right = createAndAppend('div', container, { class: 'right_div' });
     fetchJSON(url, (err, data) => {
```

Comment:
Since we know that the data we are getting is a list of repositories a more descriptive name for the `data` parameter would be `repositories`. This improves the readability of your code significantly (more so even, if you replace `val` by `repository in line 125:

```js
fetchJSON(url, (err, repositories) => {
  if (err) {
    createAndAppend('div', root, {
      html: err.message,
      class: 'alert-error',
    });
    return;
  }

  repositories.sort((a, b) => a.name.localeCompare(b.name));
  repositories.map((repository, index) =>
    createAndAppend('option', select, { value: index, html: repository.name }),
  );
  select.addEventListener('change', event => {
    const index = event.target.value;
    showRepos(left, repositories[index]);
    showContributors(right, repositories[index]);
  });
  showRepos(left, repositories[0]);
  showContributors(right, repositories[0]);
});
```